AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreConsecutiveUpdatesTests.swift

```diff
@@ -0,0 +1,331 @@
+//
+
+import XCTest
+
+import AmplifyPlugins
+import AWSPluginsCore
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSDataStoreCategoryPlugin
+
+class DataStoreConsecutiveUpdatesTests: SyncEngineIntegrationTestBase {
+
+    /// - Given: API has been setup with `Post` model registered
+    /// - When: A Post is saved with sync complete and deleted immediately
+    /// - Then: The Post should not be returned when queried for
+    func testSaveAndImmediateDelete() throws {
+        try startAmplifyAndWaitForSync()
+
+        // create a post
+        let myPost = Post(id: UUID().uuidString,
+                          title: "MyPost",
+                          content: "This is my post.",
+                          createdAt: Temporal.DateTime.now(),
+                          rating: 3,
+                          status: .published)
+
+        let saveSyncExpectation = expectation(description: "Post is saved and synced")
+        let deleteSyncExpectation = expectation(description: "Post is deleted and synced")
+
+        let hubListener = Amplify.Hub.listen(
+            to: .dataStore,
+            eventName: HubPayload.EventName.DataStore.syncReceived) { payload in
+            guard let mutationEvent = payload.data as? MutationEvent
+            else {
+                XCTFail("Can't cast payload as mutation event")
+                return
+            }
+
+            guard let post = try? mutationEvent.decodeModel() as? Post,
+                  post.id == myPost.id else {
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.create.rawValue {
+                XCTAssertEqual(post.title, "MyPost")
+                XCTAssertEqual(post.content, "This is my post.")
+                XCTAssertEqual(post.rating, 3)
+                saveSyncExpectation.fulfill()
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.delete.rawValue {
+                XCTAssertEqual(post.title, "MyPost")
+                XCTAssertEqual(post.content, "This is my post.")
+                XCTAssertEqual(post.rating, 3)
+                deleteSyncExpectation.fulfill()
+                return
+            }
+        }
+
+        guard try HubListenerTestUtilities.waitForListener(with: hubListener, timeout: 5.0) else {
+            XCTFail("Listener not registered for hub")
+            return
+        }
+
+        // save the post and wait for it to sync
+        let immediateSaveExpectation = expectation(description: "Post is saved")
+        Amplify.DataStore.save(myPost) { result in
+            switch result {
+            case .success:
+                immediateSaveExpectation.fulfill()
+            case .failure(let error):
+                XCTFail("Error: \(error)")
+            }
+        }
+        wait(for: [immediateSaveExpectation], timeout: networkTimeout)
+        wait(for: [saveSyncExpectation], timeout: networkTimeout)
```

Comment:
this waits for the save sync to occur, so this is not immediately deleting after the save